pkg/controllers/oauthclientscontroller/oauthclientscontroller_test.go

```diff
@@ -0,0 +1,533 @@
+	"context"
+	"encoding/base64"
+	"fmt"
+	"reflect"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clienttesting "k8s.io/client-go/testing"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	configv1 "github.com/openshift/api/config/v1"
+	oauthv1 "github.com/openshift/api/oauth/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	configv1listers "github.com/openshift/client-go/config/listers/config/v1"
+	fakeoauthclient "github.com/openshift/client-go/oauth/clientset/versioned/fake"
+	oauthv1listers "github.com/openshift/client-go/oauth/listers/oauth/v1"
+	routev1listers "github.com/openshift/client-go/route/listers/route/v1"
+	"github.com/openshift/library-go/pkg/oauth/oauthdiscovery"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+const (
+	masterPublicURL = "oauth-openshift.test.com"
+)
+
+type fakeSyncContext struct{}
+
+func (f fakeSyncContext) Queue() workqueue.RateLimitingInterface { return nil }
+func (f fakeSyncContext) QueueKey() string                       { return "" }
+func (f fakeSyncContext) Recorder() events.Recorder              { return nil }
+
+func newTestOAuthsClientsController(t *testing.T) *oauthsClientsController {
+	fakeClientset := fakeoauthclient.NewSimpleClientset()
+	oauthClientIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+
+	routeIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+	route := &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{Name: "oauth-openshift", Namespace: "openshift-authentication"},
+		Spec:       routev1.RouteSpec{Host: masterPublicURL},
+		Status: routev1.RouteStatus{
+			Ingress: []routev1.RouteIngress{
+				{
+					Host: masterPublicURL,
+					Conditions: []routev1.RouteIngressCondition{
+						{Type: routev1.RouteAdmitted, Status: corev1.ConditionTrue},
+					},
+				},
+			},
+		},
+	}
+	if err := routeIndexer.Add(route); err != nil {
+		t.Fatalf("got unexpected error while creating test OAuthsClientsController: %v", err)
+		return nil
+	}
+
+	ingressIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+	ingress := &configv1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
+		Spec: configv1.IngressSpec{
+			Domain: "test.com",
+			ComponentRoutes: []configv1.ComponentRouteSpec{
+				{Namespace: "openshift-authentication", Name: "oauth-openshift", Hostname: masterPublicURL},
+			},
+		},
+	}
+	if err := ingressIndexer.Add(ingress); err != nil {
+		t.Fatalf("got unexpected error while creating test ingress indexer: %v", err)
+		return nil
+	}
+
+	return &oauthsClientsController{
+		oauthClientClient: fakeClientset.OauthV1().OAuthClients(),
+		oauthClientLister: oauthv1listers.NewOAuthClientLister(oauthClientIndexer),
+		routeLister:       routev1listers.NewRouteLister(routeIndexer),
+		ingressLister:     configv1listers.NewIngressLister(ingressIndexer),
+	}
+}
+
+func Test_sync(t *testing.T) {
```

Comment:
`TestSync`, use camel-case for the other tests as well. snake-case is non-idiomatic in Go.

Reply:
oh i see, welp